flytestdlib/cache/auto_refresh_test.go

```diff
@@ -167,24 +174,27 @@ func TestCacheFour(t *testing.T) {
 		})
 		assert.NoError(t, err)
 
-		// Wait half a second for all resync periods to complete
 		// If the cache tries to enqueue the item, a panic will be thrown.
-		time.Sleep(500 * time.Millisecond)
+		fakeClock.Step(testResyncPeriod)
```

Comment:
These tests aren't perfect with how things are asynchronous... It's hard to verify that a sync actually happened. I wanted to add a unexpected sync counter to the auto refresh to use in unit tests but the constructor imo incorrectly returns the interface instead of a pointer to the struct (which should adhere to the interface). As such, the unit tests can't access anything package level.

Reply:
I wanted to fix that but I'm trying to reduce the blast radius of this change.